### DISCUSSION TITLE
Conditional operator without else

### DISCUSSION THREAD

**Discussion by doanhquang2005** [dotnet/csharplang#5073] [Language Ideas]
Ex: `if(x == 0)
x = 3;`
We can write: `x = x == 0 ? 3 : x;`
Is there a better way than the one above? Ex: `x = x != 0 ?: 3;`

**Comment by 333fred** [comment]
I've converted this to a discussion as issues are for fully-specified proposals.

**Comment by theunrepentantgeek** [comment]
I'm 100% sure proposals like this have been discussed in the past, but wasn't able to find any of them in 10m of searching. Perils of searching for `?` 😁 

IIRC, they falter on the expression `x != 3 : 3` not being well defined.

What's the value of the expression when the condition is false?

It's easy to handwave at trivial examples and infer a value, but expressions need to be well defined everywhere and things start falling apart very quickly:

``` csharp
x = x == 0 ? 3;         // Infer false case as x
var a = x == 0 ? 3;     // Should this also infer x ?
var b = x == z ? 3;     // Should we infer x or z? Why?
var c = x != z ? 3;     // Does this change the answer?
var d = test(x) ? 3;    // Do we infer x or something else?
var e = test(x, y) ? 3; // Do we infer x, y, or something else?
```

The overhead of writing the false part of the expression is low, and it avoids all of these problems.

**Comment by khm1600** [comment]
https://github.com/dotnet/csharplang/discussions/197
